lesson/narrative_and_checkpoints/narrative-and-checkpoints-template.md

```diff
@@ -1,51 +1,417 @@
+
+## Exercise 1: _What is Jest?_
+
+### Narrative:
+
+As developers, we are always trying to make our code bulletproof from tiny bugs that could completely break our applications. One process that can help us get one step closer to “unbreakable code” is something called "testing." Testing allows us to make sure that our code is behaving the way that we envisioned. Within the realm of testing, there are three major types of tests:
+
+There are three major types of tests:
+1. **Unit tests** involve testing fully isolated functions.
+2. **Integration** tests involve testing functions with dependencies, for example, a function that calls another function.
+3. **End-to-End** tests include testing the entire flow of a project, for instance, validating some type of DOM manipulation.
+
+When it comes to unit tests and integration tests, two fundamental things are needed to perform tests; test-runners and assertion libraries. Test-runners allow us to execute tests and also provide outputted test summaries after. Assertion libraries enable us to check the correctness of our code by adding exceptions to verify defined behaviors. When developing and testing in JavaScript, we can use a JavaScript testing framework called Jest. While there are many testing libraries out there, Jest provides a testing framework that focuses on simplicity. Jest’s framework provides both a test runner as well as an assertion library in one framework allowing us to do a lot right out of the box without much setup. In this lesson, we will analyze and test a language geo-location identifier web app. While doing this, we will explore the wonders of testing with Jest and its ability to expedite the project creation process. 
+
+Let's begin!
 
 <hr>
 
-## Exercise 1: _Insert exercise title here._
+## Exercise 2: _Setting up Jest_
 
 ### Narrative:
 
+Before we can begin testing our code with Jest, it must first be installed and configured. We will download Jest as a node package meaning that we first need to initiate node package manager (a.k.a npm). We will initiate  npm creating a package.json file by running the following command in our terminal:
+
+```bash
+npm init -y
+```
+
+The above syntax instructs the terminal to create our package.json file while filling in all setup questions with the default answer of "yes". Now that we have a way to manage our node packages, it is time to download and install Jest. We will complete this step with the following code:
+
+```bash
+npm install --save-dev jest
+```
+
+The following syntax instructs the terminal to download the Jest node package and save it as a developer dependency. Developer dependencies are only relevant during development and not included in the final project build for production. 
```

Comment:
> not included in the final project build for production.

This is not true. We should remove this. It's not adding any value in learning Jest.